examples/cvode/ginkgo/cv_kpr_ginkgo.cpp

```diff
@@ -0,0 +1,228 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): David J. Gardner @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Kvaerno-Prothero-Robinson ODE test problem:
+ *
+ *   [u]' = [ a  b ] [ (-1 + u^2 - r(t)) / (2u) ] + [ r'(t) / (2u) ]
+ *   [v]    [ c  d ] [ (-2 + v^2 - s(t)) / (2v) ]   [ s'(t) / (2v) ]
+ *
+ * This problem has analytical solution given by
+ *
+ *   u(t) = sqrt(1 + r(t))
+ *   v(t) = sqrt(2 + s(t))
+ *
+ * where, in this test, we use the functions
+ *
+ *   r(t) = 0.5 * cos(t)
+ *   s(t) = cos(2t)
+ * ---------------------------------------------------------------------------*/
+
+#include "cv_kpr_ginkgo.hpp"
+
+// Include integrator, matrix, linear solver, and vector headers
+#include <cvode/cvode.h>
+#include <nvector/nvector_serial.h>
```

Comment:
Im guessing this was a nice simple way to test the interface at first. Is that right @gardner48? I don't think it hurts to have it.

Reply:
I added this to have an example using the Ginkgo dense matrix (and because it has an analytical solution to check against). We should see about adding a larger example that needs a dense solver.